modules/oe_content_person/src/Entity/PersonJob.php

```diff
@@ -61,18 +61,16 @@ public function label() {
       // Define label for Default Person job.
       $label = $this->get('oe_role_name')->value;
       if (!$this->get('oe_role_reference')->isEmpty()) {
-        $label = $this->get('oe_role_reference')->entity->label();
-
+        $vocabulary = $this->get('oe_role_reference')->entity;
+        $label = \Drupal::service('entity.repository')->getTranslationFromContext($vocabulary)->label();
```

Comment:
We cannot get the translation from context here, or this could happen:

- Site is in French
- Person doesn't have a French translation, so it shows in English
- Person job will show the role label in French, and not in English

Let's do something like this:

```php
  public function label() {
    if ($this->bundle() === 'oe_default') {
      // Define label for Default Person job.
      $label = $this->get('oe_role_name')->value;
      if (!$this->get('oe_role_reference')->isEmpty()) {
        $label = $this->get('oe_role_reference')->entity->label();

        /** @var \Drupal\Core\Entity\ContentEntityBase $role_entity */
        $role_entity = $this->get('oe_role_reference')->entity;
        if ($role_entity->hasTranslation($this->language->getId())) {
          $label = $role_entity->getTranslation($this->language->getId())->label();
        }
      }

      if ($this->get('oe_acting')->value) {
        $label = $this->t('(Acting) @role', ['@role' => $label]);
      }

      if (!empty($label)) {
        return $label;
      }
    }

    return parent::label();
```

But try it out, don't just use it, I didn't test it.